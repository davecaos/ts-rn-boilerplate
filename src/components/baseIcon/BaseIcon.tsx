import React from 'react'
import {StyleProp, ViewStyle} from 'react-native'

import Icons from '../../assets/icons'
import {withTheme, IThemeProps} from '../../services/themes'

interface BaseIconProps extends IThemeProps {
  icon: keyof typeof Icons
  color?: string
  size?: number
  style?: StyleProp<ViewStyle>
}

const BaseIcon: React.FC<BaseIconProps> = ({icon, size = 24, color, style = {}, theme}) => {
  const Icon = Icons[icon] || null
  const props = {
    width: size,
    height: size,
    color: color || theme?.colors.backBlack
  }

  return Icon && <Icon {...props} style={style} />
}

export default withTheme<BaseIconProps>(BaseIcon)
