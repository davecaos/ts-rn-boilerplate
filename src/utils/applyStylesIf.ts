import {
  ViewStyle,
  FlexStyle,
  ImageStyle,
  TransformsStyle,
  TextStyle,
  ShadowStyleIOS,
  TextStyleAndroid,
  TextStyleIOS
} from 'react-native'

type applyStylesIfStylesParam =
  | ViewStyle
  | FlexStyle
  | ImageStyle
  | TransformsStyle
  | TextStyle
  | ShadowStyleIOS
  | TextStyleAndroid
  | TextStyleIOS

/**
 * @name applyStylesIf
 * @description If the condition provided in the first argument evaluates to true then apply the styles from the second argument, otherwise returns an empty object.
 */
export const applyStylesIf = (
  condition: boolean | undefined,
  styles: applyStylesIfStylesParam
): applyStylesIfStylesParam | Record<string, never> => (condition ? styles : {})
