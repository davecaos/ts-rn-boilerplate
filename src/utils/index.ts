export * from './switchCallback'
export * from './applyStylesIf'
