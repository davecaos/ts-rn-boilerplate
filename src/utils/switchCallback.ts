/**
 * @name switchCallback
 * @description Switch callback based on condition. If the condition evaluates to true, it returns the callback, otherwise returns an empty function.
 */
export const switchCallback = <C>(condition: boolean, callback: C): C | (() => void) => {
  /* eslint-disable-next-line @typescript-eslint/no-empty-function */
  if (!condition) return () => {}
  return callback
}
