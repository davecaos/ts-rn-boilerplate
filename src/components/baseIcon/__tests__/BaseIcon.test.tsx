import React from 'react'
import {render} from '../../../utils/test'
import BaseIcon from '../BaseIcon'

describe('BaseIcon', () => {
  test('Snapshot Test', () => {
    const {toJSON} = render(<BaseIcon icon="iconly_profile_outline" />, {})
    expect(toJSON()).toMatchSnapshot()
  })
})
