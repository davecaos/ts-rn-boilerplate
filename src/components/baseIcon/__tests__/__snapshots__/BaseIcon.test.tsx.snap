// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseIcon Snapshot Test 1`] = `null`;
